mne/encoding/feature.py

```diff
@@ -0,0 +1,298 @@
+
+        # Iterate through event types and create columns of event onsets.
+        events_continuous = np.zeros([self.n_times,
+                                      len(unique_event_types) + n_covs])
+        event_names = []
+        for ii, ev_type in enumerate(unique_event_types):
+            msk_events = event_ids == ev_type
+            i_ev = event_ixs[msk_events]
+            events_continuous[i_ev, ii] = 1
+
+            # Handle event names
+            event_names.append(ev_dict_rev[ev_type])
+        if n_covs > 0:
+            for iev, icov in zip(event_ixs, covs):
+                events_continuous[iev, -n_covs:] = icov
+            event_names = event_names + cov_names
+            events_continuous = sparse.csr_matrix(events_continuous)
+
+        self.names_ = event_names
+        self.unique_event_types_ = unique_event_types
+        return events_continuous
+
+    def fit_transform(self, event_ixs, event_ids=None, event_dict=None,
+                      covariates=None, covariate_names=None):
+        """Does nothing, only used for sklearn compatibility.
+
+        Parameters
+        ----------
+        X : array
+        y : array
+
+        Returns
+        -------
+        self
+        """
+        out = self.transform(
+            event_ixs, event_ids=event_ids, event_dict=event_dict,
+            covariates=covariates, covariate_names=covariate_names)
+        return out
+
+
+def delay_timeseries(ts, delays, sfreq=1.):
+    """Return a time-lagged input timeseries.
+
+    Parameters
+    ----------
+    ts: array, shape (n_times, n_feats)
+        The timeseries to delay
+    delays: array of floats, shape (n_delays) |
+            list of arrays, len == n_feats
+        The time (in seconds) of each delay for specifying
+        pre-defined delays. Negative means timepoints in the past,
+        positive means timepoints in the future. If a list, then
+        one may supply a different number of delays per feature.
+    sfreq: int
+        The sampling frequency of the series
+
+    Returns
+    -------
+    delayed: array, shape(n_times, n_feats * n_delays)
+        The delayed matrix
+    """
+    delays, sfreq = _check_delayer_params(delays, sfreq)
+    if not isinstance(delays[0], (list, np.ndarray)):
+        delays = [delays] * ts.shape[1]
+
+    delays = [np.asarray(ii) for ii in delays]
+    if len(delays) != ts.shape[1]:
+        raise ValueError('Supply either 1 set of delays, or 1 per feature')
+
+    n_times, n_feats = ts.shape
+    delays_ixs = [np.round(ii * sfreq).astype(int) for ii in delays]
+
+    # Convert ts to dense if it is sparse
+    if isinstance(ts, sparse.spmatrix):
+        is_sparse = True
+        sp_type = type(ts)
+        ts = ts.toarray()
+    else:
+        is_sparse = False
+
+    # Iterate through indices and append
+    delayed = []
+    delayed_names = []
+    for ii, (delay_rw, delay_data) in enumerate(zip(delays_ixs, ts.T)):
```

Comment:
I used to do it this way as well but I found it more difficult to follow the logic which is why I was explicitly doing it with concatenate even though it took more lines. That said, I think the way you've formulated it here is actually pretty clear. I think I am a little irrationally reshape-averse because I've found it really easy to mess up how the reindexing works